src/main/java/com/mraof/minestuck/block/machine/ComputerBlock.java

```diff
@@ -123,14 +142,72 @@ private boolean insertDisk(ComputerBlockEntity blockEntity, BlockState state, Le
 			blockEntity.setChanged();
 			level.sendBlockUpdated(pos, state, state, 3);
 			return true;
-		} else return false;
+		}
+		
+		return false;
+	}
+	
+	private boolean inputCode(ComputerBlockEntity blockEntity, BlockState state, Level levelIn, BlockPos pos, Player player, InteractionHand handIn)
+	{
+		ItemStack heldStack = player.getItemInHand(handIn);
+		if(heldStack.getItem() instanceof ReadableSburbCodeItem)
```

Comment:
Can you replace these instanceof checks with appropriate `heldStack.is(item)` checks instead?

Reply:
I have no plans of my own, but I was wondering if it would be appropriate to keep it as is in case addons wanted to add their own book versions? There may be other sections of code I cant think of right now that would prevent that but it seemed useful to avoid hardcoding

Reply:
Oh, here's a related idea: instead of handling sburb codes as part of `computerBlock.use()`, how about instead using `sburbCodeItem.onItemUseFirst()`?

Reply:
Changed, and seems to work fine in the new setting